src/components/document/annotation-layer.tsx

```diff
@@ -41,9 +42,24 @@ export const AnnotationLayer = observer(function AnnotationLayer({
   const ui = useUIStore();
   const persistentUI = usePersistentUIStore();
   const tileApiInterface = useContext(TileApiInterfaceContext);
+  const hotKeys = useRef(new HotKeys());
+
+  useEffect(() => {
+    if (!readOnly) {
+      hotKeys.current.register({
+        "delete": () => content?.deleteSelected(),
+        "backspace": () => content?.deleteSelected()
+      });
+    }
```

Comment:
I would think you'd need to return a disposer function which unregisters the handlers here. Otherwise, on a transition to `readOnly` the handlers would still be present and a subsequent transition to editable would result in redundant handlers being registered. Something like:
```suggestion
    const deleteSelected = () => content?.deleteSelected()
    if (!readOnly) {
      hotKeys.current.register({
        "delete": () => deleteSelected,
        "backspace": () => deleteSelected
      });
      return () => {
        hotKeys.current.unregister({
          "delete": () => deleteSelected,
          "backspace": () => deleteSelected
        });
      }
    }
```

Reply:
Hm, I did not think we ever changed a tile from readOnly to writable, or vice versa, during its lifetime.   But I can put  this in just for safety.